auth/eth.go

```diff
@@ -24,3 +43,33 @@ func verifyTron(tx SignedTx) ([]byte, error) {
 	}
 	return tronAddr.Bytes(), nil
 }
+
+func verifyEthereumTransacton(signedTx SignedTx) ([]byte, error) {
+	var nonceTx auth.NonceTx
+	if err := proto.Unmarshal(signedTx.Inner, &nonceTx); err != nil {
+		return nil, err
+	}
+
+	var txTx types.Transaction
+	if err := proto.Unmarshal(nonceTx.Inner, &txTx); err != nil {
+		return nil, err
+	}
+
+	var msg vm.MessageTx
+	if err := proto.Unmarshal(txTx.Data, &msg); err != nil {
+		return nil, err
+	}
+
+	var tx etypes.Transaction
+	if err := rlp.DecodeBytes(msg.Data, &tx); err != nil {
+		return nil, err
+	}
+
+	chainConfig := utils.DefaultChainConfig()
+	ethSigner := etypes.MakeSigner(&chainConfig, big.NewInt(1))
+	from, err := etypes.Sender(ethSigner, &tx)
+	if err != nil {
+		return nil, err
+	}
+	return from.Bytes(), err
```

Comment:
Also check that `tx.To()` matches `msg.To`, and that `NonceTx.Sequence` matches the `tx` nonce.